homeassistant/components/bluetooth/manager.py

```diff
@@ -235,6 +235,23 @@ def async_stop(self, event: Event) -> None:
             self._cancel_unavailable_tracking.clear()
         uninstall_multiple_bleak_catcher()
 
+    async def async_get_devices_by_address(
+        self, address: str, connectable: bool
+    ) -> list[BLEDevice]:
+        """Get devices by address."""
+        types_ = (True,) if connectable else (True, False)
+        return [
+            device
+            for device in await asyncio.gather(
+                *(
+                    scanner.async_get_device_by_address(address)
```

Comment:
Ideally in the future bleak supports this natively and we can make this a callback instead.